src/pages/core/standard-library/cryptography/p256.mdx

```diff
@@ -12,6 +12,45 @@ added to CosmWasm.
 
 ## Example
 
-## Signature verification
+### Signature verification
 
-<Callout>TODO: Add example as soon as secp256r1 is properly released</Callout>
+```rust filename="contract.rs" template="core"
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(
+    deps: Deps,
+    _env: Env,
+    msg: EcdsaVerifyMsg,
+) -> StdResult<QueryResponse> {
+    let public_key = msg.public_key;
+    let signature = msg.signature;
+    let message_hash = msg.message_hash;
+
+    // Verify the signature. On chain!
+    let is_valid = deps.api.secp256r1_verify(&message_hash, &signature, &public_key)?;
+    let response = format!("{{ \"is_valid\": {is_valid} }}");
+
+    Ok(QueryResponse::new(response.into_bytes()))
+}
+```
+
+### Public key recovery
+
+```rust filename="contract.rs" template="core"
+#[cfg_attr(not(feature = "library"), entry_point)]
+pub fn query(
+    deps: Deps,
+    _env: Env,
+    msg: EcdsaRecoverMsg,
+) -> StdResult<QueryResponse> {
+    let signature = msg.signature;
+    let message_hash = msg.message_hash;
+    let recovery_id = msg.recovery_id;
+
+    // Recover the public key. On chain!
+    let public_key = deps.api.secp256r1_recover_pubkey(&message_hash, &signature, recovery_id)?;
+    let public_key = HexBinary::from(public_key).to_hex();
```

Comment:
Here typed responses are even better since HexBinary is primarily made for hex encoding in serde